todomvc.js

```diff
@@ -1,102 +1,91 @@
+            />
+            <label htmlFor="toggle-all">Mark all as complete</label>
+        <List todos={ todosList ? [...todosList] : [] }  />   
+        </section>
+        <footer id="footer">
+            <span id="todo-count">
+                <strong>{todosList && todosList.active.length}</strong> items left
+            </span>
 
-				<section id="main">
-					<input
-						id="toggle-all"
-						type="checkbox"
-						checked={ this.viewModel.allChecked ? true : false }
-						disabled={ this.viewModel.todosList && this.viewModel.todosList.saving.length }
-					/>
-					<label htmlFor="toggle-all">Mark all as complete</label>
-					<List todos={ this.viewModel.todosList } />
-				</section>
+            <ul id="filters">
+                <li>
+                    <a
+                        href={route.url({ filter: undefined })}
+                        className={ route.current({ filter: undefined }) ? "selected" : "" }
+                        >All</a>
+                </li>
+                <li>
+                    <a
+                        href={route.url({ filter: "active" })}
+                        className={ route.current({ filter: "active" }) ? "selected" : "" }
+                        >Active</a>
+                </li>
+                <li>
+                    <a
+                        href={route.url({ filter: "complete" })}
+                        className={ route.current({ filter: "complete" }) ? "selected" : "" }
+                        >Completed</a>
+                </li>
+            </ul>
 
-				<footer id="footer">
-					<span id="todo-count">
-						<strong>{this.viewModel.todosList && this.viewModel.todosList.active.length}</strong> items left
-					</span>
+            <button id="clear-completed"
+                onClick={ () => todosList.destroyComplete() }>
+                Clear completed ({todosList && todosList.complete.length})
+            </button>
+        </footer>
+    </section>
+);}
 
-					<ul id="filters">
-						<li>
-							<a
-								href={route.url({ filter: undefined })}
-								className={ route.current({ filter: undefined }) ? "selected" : "" }
-								>All</a>
-						</li>
-						<li>
-							<a
-								href={route.url({ filter: "active" })}
-								className={ route.current({ filter: "active" }) ? "selected" : "" }
-								>Active</a>
-						</li>
-						<li>
-							<a
-								href={route.url({ filter: "complete" })}
-								className={ route.current({ filter: "complete" }) ? "selected" : "" }
-								>Completed</a>
-						</li>
-					</ul>
+const App = connect(Store)(TodoMVC);
+export default App;
 
-					<button id="clear-completed"
-						onClick={ () => this.viewModel.todosList.destroyComplete() }>
-						Clear completed ({this.viewModel.todosList && this.viewModel.todosList.complete.length})
-					</button>
-				</footer>
-			</section>
-		);
-	}
-}
-
-App.ViewModel = ViewModel;
+route.data = Store;
```

Comment:
I'm not sure this will actually work.  You'll need a single instance of `Store`.  Is the routing actually working?

Reply:
There are a couple of ways to bind ylem to can-route. The way it was done in the previous version of this example was like this, (with `viewModel` changed for `props`). I have also done it in the `componentDidMount` lifecycle callback, with the same contents of teh function below called there instead.

```javascript
ReactDOM.render(<App ref={register} />, div);

function register(app) {
	route.data = app.props;
	route.ready();
}
```

Another idea would be to put the route data as external to the component and pass it in as a prop. As long as route.data is an observable (which I assume is a requirement of can-route), ylem will see those same updates. If App is not `connect`ed, this would be a use-case for `observe`

```javascript
route.data = new RouteStore();
route.ready();

ReactDOM.render(<App route={route.data} />, div);
```